src/pip/_internal/cli/base_command.py

```diff
@@ -213,6 +213,8 @@ def _main(self, args):
                 "This will become an error in pip 21.0."
             )
 
+        self.handle_pip_version_check(options)            
```

Comment:
Hmm... I'm wary of doing it like this -- if this consistently fails or raises an error, then the user would basically not be able to use pip in any way.

Reply:
Agreed. If (for example) a user has no internet, they should still be able to use pip offline. A network error in the check shouldn't kill pip.

We should probably:

1. Catch and suppress any errors from this check.
2. Add tests to ensure that we did so successfully.

Maybe the contract for `handle_pip_version_check` should simply say that it will never generate an exception? I've been playing with Rust recently, so my brain translates this as "the function should just not return a `Result`" but it's harder to enforce such constraints in Python 🙁

Reply:
FWIW, I'm also not sure that this is the correct thing to do TBH. The message being at the end of the output is a good thing when you have an install run that spewed out a lot of data.

Reply:
TBH, I don't have a strong opinion about this either way. I suspect that whatever we do with the message, there will always be a certain proportion of people who hate it, so I don't think it's worth putting a lot of effort into improving it. Ultimately, it does the job of making people aware of the need to keep pip updated, and that's the key thing for me.